layers/ospf.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2017 Google, Inc. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+package layers
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/google/gopacket"
+)
+
+type OSPFType uint8
```

Comment:
Please add comments for all new types, and for exported functions (even if they are simply "DecodeFromBytes implements gopacket.DecodingLayer" or the like).